viamonvif/device/device_test.go

```diff
@@ -91,3 +92,106 @@ func readBody(r *http.Request) (string, error) {
 	}
 	return buf.String(), nil
 }
+
+func TestTLSVerificationConfig(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := readBody(r)
+		if err != nil {
+			t.Fatalf("failed to read body: %v", err)
+		}
+
+		if strings.Contains(r.Header.Get("Content-Type"), "soap") &&
+			strings.Contains(body, "GetCapabilities") {
+			w.Write([]byte(`<?xml version="1.0" encoding="UTF-8"?>
+				<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2003/05/soap-envelope">
+					<SOAP-ENV:Body>
+						<GetCapabilitiesResponse>
+							<Capabilities>
+								<Media>
+									<XAddr>http://example.com/onvif/media</XAddr>
+								</Media>
+							</Capabilities>
+						</GetCapabilitiesResponse>
+					</SOAP-ENV:Body>
+				</SOAP-ENV:Envelope>`))
+		}
+	}))
+	defer server.Close()
```

Comment:
We should also test that the close does not error.. You can defer that entire test case.

Reply:
Oh wait I think server.Close() doesn't return error

Reply:
Blergh, okay.